build.gradle

```diff
@@ -89,5 +89,9 @@ subprojects {
     installDist {
         duplicatesStrategy DuplicatesStrategy.EXCLUDE
     }
+
+    configurations.all {
+       resolutionStrategy.force "org.slf4j:slf4j-api:${slf4jVersion}", "org.slf4j:slf4j-simple:${slf4jVersion}"
```

Comment:
Instead of this, how about excluding the `slf4j` dependency from the `mariadb` dependency?

Reply:
Agreed. I think excluding it is a bit more polite way.

Reply:
Yes, we can I.
I thought that using the `resolutionStrategy.force` was future proof in case another library we update pulls `sl4j` v2.

Reply:
The Gradle doc explains that using the `resolutionStrategy.force` is a project-global excluding rule.

```
Excluding a particular transitive dependency does not guarantee that it does not show up in the dependencies of a given configuration. For example, some other dependency, which does not have any exclude rules, might pull in exactly the same transitive dependency. To guarantee that the transitive dependency is excluded from the entire configuration please use per-configuration exclude rules: [getExcludeRules](https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.artifacts/-configuration/get-exclude-rules.html). In fact, in a majority of cases the actual intention of configuring per-dependency exclusions is really excluding a dependency from the entire configuration (or classpath).

If your intention is to exclude a particular transitive dependency because you don't like the version it pulls in to the configuration then consider using forced versions' feature: [force](https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.artifacts/-resolution-strategy/force.html).
```
cf. https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.artifacts/-module-dependency/exclude.html

Could you share your concerns concerning the `resolutionStrategy.force` ?
I am ok but I just want to better understand your point of view.

Reply:
The 2 ways have similar effects, and actually either is fine with me. IIUC, the slight difference is basically only global-scope or not. So, this might be also a matter of taste thing. I relatively prefer starting with small-scope effect, but I don't insist on it as I understand the pros of `resolutionStrategy.force`.

Reply:
I was concerned about the side effects of `resolutionStrategy.force` since it’s a project-global configuration. Also, I believe we’ve usually used the `exclude` configuration for similar issues in the past. Do you think `resolutionStrategy.force` is a better approach than the `exclude` configuration?

Reply:
My understanding was that `resolutionStrategy.force` could be used as project-wide `exclude` which means:
1.  The forced dependency version was used project-wide when building ScalarDB
2. Other projects using ScalarDB as a dependency would also be presented with the forced dependency version, i.e. excluding the non-wanted version

But after testing locally, statement 2. is wrong. `slf4j-api:2.0.7` was still listed as a transitive dependency of  `com.github.waffle:waffle-jna`.

In comparison, when using the `exclude` syntax. `slf4j-api:2.0.7` was not listed as a dependency or pulled by `com.github.waffle:waffle-jna`, which is what we want to avoid the problem described by Toshi in https://github.com/scalar-labs/scalardb/pull/2420#issuecomment-2552885802

So, I will revise the PR to use the exclude syntax.